### ISSUE TITLE
nc_rename_var() fails to rename non-coordinate to coordinate

### ISSUE THREAD

**Issue by czender** [Unidata/netcdf-c#381]
### Environment Information

* What platform are you using? (please provide specific distribution/version in summary)
  * [x] Linux
  * [ ] Windows
  * [x ] OSX
  * [ ] Other
  * [ ] NA
* Can you provide a sample netCDF file or `C` code to recreate the issue?
  * [x] Yes (please attach to this issue, thank you!)
  * [ ] No
  * [ ] Not at this time

### Summary of Issue

As pointed out by ncrename user Paolo Oliveri, netCDF4 fails to rename a non-coordinate variable to a coordinate variable.

### Steps to reproduce the behavior

ncrename -v non_coord,coord in_grp.nc foo.nc # Fails (HDF error)

Input file in_grp.nc is [here](http://dust.ess.uci.edu/tmp/in_grp.nc)

Full errors from logging:

```
zender@skyglow:~$ ncrename -O -v non_coord,coord ~/nco/data/in_grp.nc ~/foo.nc
HDF5-DIAG: Error detected in HDF5 (1.8.17) thread 0:
  #000: ../../src/H5Gdeprec.c line 509 in H5Gmove(): couldn't move link
    major: Links
    minor: Unable to initialize object
  #001: ../../src/H5Gdeprec.c line 591 in H5G_move(): unable to move link
    major: Links
    minor: Can't move object
  #002: ../../src/H5L.c line 2704 in H5L_move(): unable to find link
    major: Symbol table
    minor: Object not found
  #003: ../../src/H5Gtraverse.c line 861 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #004: ../../src/H5Gtraverse.c line 641 in H5G_traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #005: ../../src/H5L.c line 2564 in H5L_move_cb(): unable to follow symbolic link
    major: Symbol table
    minor: Object not found
  #006: ../../src/H5Gtraverse.c line 861 in H5G_traverse(): internal path traversal failed
    major: Symbol table
    minor: Object not found
  #007: ../../src/H5Gtraverse.c line 641 in H5G_traverse_real(): traversal operator failed
    major: Symbol table
    minor: Callback failed
  #008: ../../src/H5L.c line 2425 in H5L_move_dest_cb(): an object with that name already exists
    major: Symbol table
    minor: Object not found
nco_err_exit(): ERROR Short NCO-generated message (usually name of function that triggered error): nco_rename_var()
nco_err_exit(): ERROR Error code is -101. Translation into English with nc_strerror(-101) is "NetCDF: HDF error"
nco_err_exit(): ERROR NCO will now exit with system call exit(EXIT_FAILURE)

```

**Comment by WardF** [comment]
Just to clarify, is this an issue where something used to work but now doesn't, or is this a brand new thing that nobody has tried before to our knowledge?

**Comment by czender** [comment]
This specific renaming has not been tried before. My guess is that the bug is not new, and is related to the underlying rename problems that have been reported before and are documented [here](http://nco.sf.net/nco.html#ncrename_crd).

**Comment by WardF** [comment]
Ok that saves me some time trying to determine when it broke. I'll take a look at it as quickly as I can, thanks!

**Comment by edhartnett** [comment]
I believe this issue is fixed in the 4.6.0 release. @czender if so, please close this ticket.
